android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt

```diff
@@ -101,11 +101,14 @@ open class QuestionnaireActivity : BaseMultiLanguageActivity(), View.OnClickList
         ?: emptyList()
 
     prePopulationParams =
-      actionParams.filter {
-        it.paramType == ActionParameterType.PREPOPULATE &&
-          !it.value.isNullOrEmpty() &&
-          !it.value.contains(STRING_INTERPOLATION_PREFIX)
-      }
+      actionParams
+        .asSequence()
+        .filter {
+          it.paramType == ActionParameterType.PREPOPULATE &&
+            !it.value.isNullOrEmpty() &&
+            !it.value.contains(STRING_INTERPOLATION_PREFIX)
+        }
+        .toList()
```

Comment:
Is this `toList` necessary? We only have a single filter operation in this sequence.

Reply:
@ellykits it might be needed since `prePopulationParams` is of type `List<ActionParamater>`. However, using a sequence converts the collection to a `Sequence<ActionParameter>`. Maybe we can ignore the asSequence() or convert it toList() before reading elements or traversing within the collection